src/main/groovy/com/marklogic/gradle/task/CorbTask.groovy

```diff
@@ -7,64 +7,166 @@ import com.marklogic.appdeployer.AppConfig
 
 class CorbTask extends JavaExec {
 
-    String xccUrl
-    String collectionName = '""'
-
-    // It's common practice for the uris/transform modules to have the same prefix, so just set this if that's the
-    // case - e.g. convert-uris.xqy and convert-transform.xqy
-    String modulePrefix
-
-    // Otherwise, set transformModule and urisModule
-    String transformModule
-    String urisModule
-    
-    // corb defaults to 1, but 8 seems like a more common default
-    int threadCount = 8
-    
-    String moduleRoot = "/"
-    
-    String modulesDatabase
-    
-    String install = "false"
-    
-    @TaskAction
-    @Override
-    public void exec() {
-        setMain("com.marklogic.developer.corb.Manager")
-
-        AppConfig config = getProject().property("mlAppConfig")
-
-        List<String> newArgs = new ArrayList<>()
-
-        newArgs.add(xccUrl)
-
-        newArgs.add(collectionName)
-
-        if (modulePrefix) {
-            newArgs.add(modulePrefix + "-transform.xqy")
-        } else {
-            newArgs.add(transformModule)
-        }
-        
-        newArgs.add(threadCount + "")
-        
-        if (modulePrefix) {
-            newArgs.add(modulePrefix + "-uris.xqy")
-        } else {
-            newArgs.add(urisModule)
-        }
-        
-        newArgs.add(moduleRoot)
-        
-        if (modulesDatabase) {
-            newArgs.add(modulesDatabase)
-        } else {
-            newArgs.add(config.getModulesDatabaseName())
-        }
-        
-        newArgs.add(install)
-        
-        setArgs(newArgs)
-        super.exec()
+  // prefix for corb project properties, to ensure no conflicts with other project properties
+  private static final String CORB_PROPERTY_PREFIX = "corb"
+
+  String xccUrl //same as xccConnectionUri
+  // It's common practice for the uris/transform modules to have the same prefix, so just set this if that's the
+  // case - e.g. convert-uris.xqy and convert-transform.xqy
+  String modulePrefix
+  // Otherwise, set processModule (formerly transformModule) and urisModule
+  String transformModule //same as processModule
+
+  def install = false // INSTALL
+
+  String moduleRoot = "/"  // MODULE-ROOT
+
+  // corb defaults to 1, but 8 seems like a more common default
+  def threadCount = 8 // THREAD-COUNT
+
+  @TaskAction
+  @Override
+  public void exec() {
+    setClasspath(getProject().configurations.corb)
```

Comment:
Yes, makes sense. We could take it out. Or what about conditionally setting the classpath if there is a configuration named 'corb'?

```
if (getProject().configurations.findByName('corb')) {
  setClasspath(getProject().configurations.corb)
}
```